mslib/mscolab/file_manager.py

```diff
@@ -250,6 +256,58 @@ def modify_user(self, user, attribute=None, value=None, action=None):
             db.session.commit()
         return True
 
+    def upload_file(self, file, subfolder, identifier=None):
+        """
+        Generic function to save files securely in specified directory with unique filename
+        and return the relative file path.
+        """
+        upload_folder = mscolab_settings.UPLOAD_FOLDER
+        with fs.open_fs(upload_folder) as _fs:
+            file_dir = fs.path.join(upload_folder, str(subfolder) if subfolder else "")
+            if sys.platform.startswith('win'):
+                file_dir = file_dir.replace('\\', '/')
+            _fs.makedirs(str(subfolder) if subfolder else "", recreate=True)
+
+            # Creating unique and secure filename
+            file_name, _ = file.filename.rsplit('.', 1)
+            mime_type, _ = mimetypes.guess_type(file.filename)
+            file_ext = mimetypes.guess_extension(mime_type) if mime_type else '.unknown'
+            token = secrets.token_urlsafe()
+            timestamp = time.strftime("%Y%m%dT%H%M%S")
+
+            if identifier:
+                file_name = f'{identifier}-{timestamp}-{token}{file_ext}'
+            else:
+                file_name = f'{file_name}-{timestamp}-{token}{file_ext}'
+            file_name = secure_filename(file_name)
+
+            # Saving the file
+            file_path = fs.path.join(file_dir, file_name)
+            file.save(file_path)
+
+            # Relative File path
+            upload_folder = upload_folder.replace('\\', '/')
+            static_dir = fs.path.basename(upload_folder)
+            static_file_path = os.path.join(static_dir, str(subfolder), file_name)
+            static_file_path = static_file_path.replace('\\', '/')
+            logging.debug(f'Relative Path: {static_file_path}')
```

Comment:
This again mixes `fs` and `os`. This could be simplified to
```python
fs.path.relativefrom(upload_folder, fs.path.join(<what was supplied to open_fs>, file_name))
```
when the subfolder is opened as `_fs`.

Reply:
I refactored the code according to this, now the static path is returned is relative to the uploads directory, which works fine for profile images,
BUT, as this function also handles chat attachment functionality, the chat attachments now return a path like 
`6/{filename}`, which breaks the code for adding the messages to the chat as it expects a path like `uploads/6/{filename}`
Please suggest me what I must do.

Reply:
Hmm, without seeing the exact error I can only guess, but if the chat attachment stuff breaks then that code has the same bug and also needs to be fixed.

But that is a bit more work: it would require a database migration to fix the wrong paths that are already saved and the migration infrastructure to do that properly doesn't exist yet. Also, that would be well out of scope for your project.

I think you could just introduce a keyword argument to specify if the method returns the path with or without the prefix to allow the chat attachment code to get the old wrongly-prefixed path, while using the proper path for profile images. The default should be to not add the prefix.

That can then be fixed and refactored later.

@ReimarBauer does that sound reasonable or do you have a different idea?

Reply:
I'm fine with this solution.

Reply:
@workaryangupta please add also a ToDo here

ToDo: add a namespace for the chat, similar as for profile

Reply:
with a manually migration the existing chat message attachment can be seen. so that needs no additional migration, but when we implement the namespace it definitly does.